interfaces/builtin/modem_manager.go

```diff
@@ -0,0 +1,1205 @@
+
+dbus (send)
+   bus=system
+   path=/org/freedesktop/DBus
+   interface=org.freedesktop.DBus
+   member={Request,Release}Name
+   peer=(name=org.freedesktop.DBus),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="org.freedesktop.ModemManager1",
+
+# Allow traffic to/from our path and interface with any method
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/ModemManager1{,/**}
+    interface=org.freedesktop.ModemManager1*,
+
+# Allow traffic to/from org.freedesktop.DBus for ModemManager service
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/ModemManager1{,/**}
+    interface=org.freedesktop.DBus.*,
+`)
+
+var modemManagerConnectedPlugAppArmor = []byte(`
+# Description: Allow using ModemManager service. Reserved because this gives
+#  privileged access to the ModemManager service.
+# Usage: reserved
+
+#include <abstractions/dbus-strict>
+
+# Allow all access to ModemManager service
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/ModemManager1{,/**}
+    peer=(label=###SLOT_SECURITY_TAGS###),
+`)
+
+var modemManagerPermanentSlotSecComp = []byte(`
+# Description: Allow operating as the ModemManager service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+accept
+accept4
+bind
+connect
+getpeername
+getsockname
+getsockopt
+listen
+recv
+recvfrom
+recvmmsg
+recvmsg
+send
+sendmmsg
+sendmsg
+sendto
+setsockopt
+shutdown
+socketpair
+socket
+`)
+
+var modemManagerConnectedPlugSecComp = []byte(`
+# Description: Allow using ModemManager service. Reserved because this gives
+#  privileged access to the ModemManager service.
+# Usage: reserved
+
+# Can communicate with DBus system service
+connect
+getsockname
+recv
+recvmsg
+recvfrom
+send
+sendto
+sendmsg
+socket
+`)
+
+var modemManagerPermanentSlotDBus = []byte(`
+<!-- DBus policy for ModemManager (upstream version 1.4.0)
+  - TODO change to polkit one when we enable it -->
+<policy context="default">
+    <allow send_destination="org.freedesktop.ModemManager1"/>
+</policy>
+
+<policy user="root">
+    <allow own="org.freedesktop.ModemManager1"/>
+</policy>
+`)
+
+var modemManagerPermanentSlotUdev = []byte(`
+# Concatenation of all ModemManager udev rules
+# do not edit this file, it will be overwritten on update
+
+ACTION!="add|change|move", GOTO="mm_cinterion_port_types_end"
+SUBSYSTEM!="tty", GOTO="mm_cinterion_port_types_end"
+ENV{ID_VENDOR_ID}!="1e2d", GOTO="mm_cinterion_port_types_end"
+
+SUBSYSTEMS=="usb", ATTRS{bInterfaceNumber}=="?*", ENV{.MM_USBIFNUM}="$attr{bInterfaceNumber}"
+
+ATTRS{idVendor}=="1e2d", ATTRS{idProduct}=="0053", ENV{.MM_USBIFNUM}=="01", ENV{ID_MM_CINTERION_PORT_TYPE_GPS}="1"
+
+LABEL="mm_cinterion_port_types_end"
+# do not edit this file, it will be overwritten on update
+
+ACTION!="add|change|move", GOTO="mm_mbm_end"
+SUBSYSTEMS=="usb", GOTO="mm_mbm_check"
+GOTO="mm_mbm_end"
+
+LABEL="mm_mbm_check"
+
+# Ericsson F3507g
```

Comment:
Not blocking on the unimplemented idea I posted above, but if modem-manager can use the new way of looking for vendor/product ID in udev events, I think the dynamically generated apparmor rules for udev files would allow this to work and the old udev rules way of doing things can go away?